src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int AMOUNT_INDEX = 1;
+    private static final int TYPE_INDEX = 0;
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String text = readFile(fromFileName);
+        writeFile(toFileName, text);
+    }
+
+    private String readFile(String fromFile) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        int resultAmount;
+        String report;
+        StringBuilder builder = new StringBuilder();
```

Comment:
The `builder` variable could have a more informative name. Consider renaming it to something like `reportBuilder`. This is related to the 'Remember about informative names of the variables' item from the checklist.